features/support/env.rb

```diff
@@ -0,0 +1,54 @@
+require 'fileutils'
+require 'socket'
+require 'timeout'
+
+testBuildFolder = 'features/fixtures/testbuild'
+
+Dir.mkdir testBuildFolder
+
+# Copy the existing air
+`find . -name '*.go' \
+        -not -path "./examples/*" \
+        -not -path "./testutil/*" \
+        -not -path "./features/*" \
+        -not -name '*_test.go' | cpio -pdm #{testBuildFolder}`
+
+
+# Scenario hooks
+Before do
+# Runs before every Scenario
+end
+
+After do
+# Runs after every Scenario
+end
+
+at_exit do
+# Runs when the test run is completed
+  FileUtils.rm_rf(testBuildFolder)
+end
+
+def port_open?(ip, port, seconds=1)
```

Comment:
I was trying detecting when the port is open rather than pinging a request as this was causing a session to be created and invalidating the tests. I'm not sure if this is working or not, but I'll push it upstream to maze runner if I can get it working

Reply:
Upstream PR: https://github.com/bugsnag/maze-runner/pull/48